pkg/client/logs_test.go

```diff
@@ -201,3 +203,76 @@ func TestLogReaderNoError(t *testing.T) {
 		})
 	}
 }
+
+func getFakeKubernetesClientSet() kubernetes.Interface {
+	// TODO(ashish-amarnath): figure out how to fake this
```

Comment:
You and me both ;)

Many functions use the `kubernetes.Interface` because it is the top level client but in general it seems un-fakeable because it is such a large API. Instead you have to try to concentrate as much of your logical code into a testable set of functions which use a much smaller interface.

Reply:
I finally figured out how to do this and it was always apparently a solved problem. Client-go creates a fake interface object and you can create actions that respond to different API calls. We now have an example in the codebase to model tests that need this functionality: https://github.com/heptio/sonobuoy/blob/6d02613d2fcbf91f284918cf75c94c9d4e00df1c/pkg/plugin/aggregation/run_test.go#L131-L138